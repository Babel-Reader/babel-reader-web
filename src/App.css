.react-pdf__Page__canvas {
    margin:auto;
    position: absolute;
    right: 0;
    left: 0;
}


.react-pdf__Page__textContent {
    height: 100% !important;
    position: absolute !important;
    transform: none !important;
    left: 0 !important;
    right: 0 !important;
    margin: auto !important;
}

