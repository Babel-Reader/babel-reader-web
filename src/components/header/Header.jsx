import React, {useContext} from "react";
import './Header.scss'
import {useDropzone} from "react-dropzone";
import {Autocomplete} from "@material-ui/lab";
import {languageList} from "../../utils";
import TextField from "@material-ui/core/TextField";
import {BookContext} from "../home/Home";

export default () => {
  const {
    file,
    setFile,
    languages,
    setLanguages,
  } = useContext(BookContext);

  const {getRootProps, getInputProps} = useDropzone({
    onDrop: (acceptedFiles) => {
      acceptedFiles[0] && setFile(acceptedFiles[0]);
    },
    accept: '.pdf'
  });

  return (
    <header className='home-header'>
      <section {...getRootProps({className: 'dropzone'})}>
        <input {...getInputProps()} />
        <div>

        </div>
        <div className='header-title'>
          <p className='header-book-title'>
            {file ? (typeof file === 'string' ? file : file.path) : ''}
          </p>
          <i>Click to open a book</i>
        </div>
      </section>

      <div className="lang-input">
        <Autocomplete
          className="lang-combo-box"
          options={languageList}
          value={languages.in}
          onChange={(e, lang) => setLanguages(
            {
              ...languages,
              in: lang
            }
          )}
          getOptionLabel={(option) => option.key}
          renderInput={(params) => <TextField {...params} label="Inlang" variant="outlined"/>}
        />
        <Autocomplete
          className="lang-combo-box"
          options={languageList}
          value={languages.out}
          onChange={(e, lang) => setLanguages({
            ...languages,
            out: lang,
          })}
          getOptionLabel={(option) => option.key}
          renderInput={(params) => <TextField {...params} label="Outlang" variant="outlined"/>}
        />
      </div>

    </header>
  );
};
