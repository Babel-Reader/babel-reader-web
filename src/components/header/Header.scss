.home-header {
  background-color: azure;
  display: flex;
  justify-content: space-between;
  padding: 0.5rem;

  .dropzone {
    width: 100%;
    margin-right: 1rem;
    padding: 0 0.5rem;
    border-style: dashed;
    border-color: #eeeeee;
    background-color: #fafafa;
    .header-title {
      display: flex;
      justify-content: space-between;
      align-items: center;
    }

  }

  .lang-input {
    display: flex;
    width: 25%;
    :first-child{
      margin-right: 1rem;
    }

    .lang-combo-box {
      flex-grow: 1;
    }
  }
}


