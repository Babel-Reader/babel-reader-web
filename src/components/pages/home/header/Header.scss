.home-header {

  .header-section{
    display: flex;
    flex-grow: 1;
  }

  .right {
    justify-content: flex-end;
  }

  .MuiToolbar-root {
    justify-content: space-between;
  }

  .dropzone {
    background-color: #00000032;
    border-color: #00000064;
    color: black;
    margin-left: 1rem;
    padding: 0.5rem ;
  }

  .lang-input {
    display: flex;
    min-width: 20rem;
    .lang-combo-box{
      margin-left: 1rem;
      width: inherit;
      flex-grow: 1;
    }
  }
}
