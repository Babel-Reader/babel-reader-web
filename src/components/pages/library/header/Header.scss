
.library-header{
  display: flex;
  flex-grow: 1;
  .MuiToolbar-root{
    justify-content: space-between;
  }
}
