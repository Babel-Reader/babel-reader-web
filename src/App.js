import React, { useEffect, useState } from 'react';
import './App.css';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './components/pages/home/Home';
import { pdfjs } from 'react-pdf';
import Library from './components/pages/library';
import { bookTitle } from 'utils';
import { auth } from 'services/firebase/firebase';

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

export const LANG_AUTO = {
  key: '??',
  name: 'auto detect',
  nativeName: 'auto detect',
};

let languageList = [
  LANG_AUTO,
  {
    key: 'en',
    name: 'English',
    nativeName: 'English',
  },
  {
    key: 'fr',
    name: 'French',
    nativeName: 'Français',
  },
  {
    key: 'es',
    name: 'Spanish',
    nativeName: 'Español',
  },
];

const fetchLanguageList = () => {
  return fetch(process.env.REACT_APP_TRANSLATE_LANGUAGE_LIST_URL)
    .then((res) => res.json())
    .then((res) => {
      if (res && res.translation) {
        const langValueForKey = (key) => {
          return {
            key,
            name: res.translation[key].name,
            nativeName: res.translation[key].nativeName,
          };
        };

        languageList = [
          LANG_AUTO,
          langValueForKey('en'),
          langValueForKey('fr'),
          langValueForKey('es'),
        ];
        Object.keys(res.translation).forEach((key) => {
          if (key !== 'en' && key !== 'fr' && key !== 'es') {
            languageList.push(langValueForKey(key));
          }
        });
      }
    });
};

export const BookContext = React.createContext({});

function App() {
  const [file, setFile] = useState();
  const [user, setUser] = useState(auth.currentUser);
  const [languages, setLanguages] = useState({
    in: LANG_AUTO,
    out: languageList[2],
  });

  useEffect(() => {
    const title = bookTitle(file);
    const pageTitle = 'Babel Reader';
    document.title = title ? `${pageTitle}: ${title}` : pageTitle;
  }, [file]);

  useEffect(() => {
    fetchLanguageList().then(() => {
      setLanguages({
        in: LANG_AUTO,
        out: languageList[2],
      });
    });
  }, []);

  return (
    <BookContext.Provider
      value={{
        file,
        setFile,
        languages,
        setLanguages,
        languageList,
        user,
        setUser,
      }}
    >
      <Router>
        <Switch>
          <Route path="/reading/samples/:bookName">
            <Home isSample />
          </Route>
          <Route path="/reading/:bookName">
            <Home />
          </Route>
          <Route>
            <Library />
          </Route>
        </Switch>
      </Router>
    </BookContext.Provider>
  );
}

export default App;
