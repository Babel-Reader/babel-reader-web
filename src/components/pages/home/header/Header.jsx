import React, { useContext, useState } from 'react';
import './Header.scss';
import { BookContext } from 'App';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import { ArrowBack, Settings } from '@material-ui/icons';
import OptionsModal from 'components/pages/home/header/OptionsModal';
import LangInput from 'components/LangInput';

export default ({history}) => {
  const [dialogOpen, setDialogOpen] = useState(false)
  const { languages, setLanguages, languageList, user, setUser } = useContext(BookContext);

  return (
    <AppBar position='static' className='home-header'>
      <Toolbar>
        <div className='header-section left'>
          <IconButton
            onClick={() => history.push('/')}
          >
            <ArrowBack/>
          </IconButton>
          <div className="lang-input header-section">
            <LangInput
              {...{
                label: 'Book Language',
                value: languages.in,
                onChange: (e, lang) => {
                  setLanguages({
                    ...languages,
                    in: lang,
                  });
                },
              }}
            />

            <LangInput
              {...{
                label: 'Translation Language',
                value: languages.out,
                onChange: (e, lang) => {
                  setLanguages({
                    ...languages,
                    out: lang,
                  });
                },
                options: languageList.slice(1),
              }}
            />
          </div>
        </div>

        <IconButton onClick={() => setDialogOpen(true)}>
          <Settings/>
        </IconButton>
        <OptionsModal
          {...{
            open: dialogOpen,
            setOpen: setDialogOpen,
            user,
            setUser,
          }}
        />

      </Toolbar>
    </AppBar>
  );
};
