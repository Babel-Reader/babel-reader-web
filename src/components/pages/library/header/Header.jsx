import React, { useContext } from 'react';
import Button from '@material-ui/core/Button';
import { BookContext } from 'App';
import Login from 'components/login';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import './Header.scss'


export default ({history})=>{
  const { user, setUser } = useContext(BookContext);

  return (
    <AppBar position='static' className='library-header'>
      <Toolbar>
        <Button className='library-header-title' onClick={()=>history.go(0)}>Library</Button>
        <Login {...{ user, setUser }}/>
      </Toolbar>
    </AppBar>
  )
}
