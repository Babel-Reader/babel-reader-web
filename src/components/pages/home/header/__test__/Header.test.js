import React from 'react';
import { create } from 'react-test-renderer';
import { BookContext } from 'App'
import Header from '../Header';


let languageList = [
  {
    key: 'auto',
    name: 'auto detect',
    nativeName: 'auto detect',
  },
  {
    key: 'en',
    name: 'English',
    nativeName: 'English',
  },

  {
    key: 'fr',
    name: 'French',
    nativeName: 'Français',
  },
  {
    key: 'es',
    name: 'Spanish',
    nativeName: 'Español',
  },
];


it('can match snapshot', () => {
  expect(
    create(
      <BookContext.Provider
        value={{
          file: 'testFile',
          languageList,
          languages: {
            in: languageList[0],
            out: languageList[2]
          },
        }}
      >
        <Header />
      </BookContext.Provider>
    )
  ).toMatchSnapshot();
});
