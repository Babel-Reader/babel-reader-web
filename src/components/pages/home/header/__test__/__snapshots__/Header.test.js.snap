// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can match snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary home-header MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="header-section left"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <div
        className="lang-input header-section"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root lang-combo-box MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              id="undefined-label"
            >
              Book Language
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root lang-combo-box MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              id="undefined-label"
            >
              Translation Language
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
            transform="scale(1.2, 1.2)"
          />
        </svg>
      </span>
    </button>
  </div>
</header>
`;
